CenterForOpenScience/ember-osf-web
# [ENG-5019] Add disease and data-type options to collection metadata

-   Ticket: [ENG-5019]
-   Feature flag: n/a

## Purpose
- Add "data-type" and "disease" as metadata options for collection submission

## Summary of Changes
- Add "data-type" and "disease" to all the places we have collection submission metadata options

## Screenshot(s)
- Collection Submit page
![image](https://github.com/CenterForOpenScience/ember-osf-web/assets/51409893/0f98e05f-170e-406d-a8b2-4c2d2167e93d)


## Side Effects
- a bit of sadness seeing all this non-DRY code

## QA Notes
- Will need to create a collection provider with options for these new fields in the admin app. This should be testable on whatever branch the backend PR for this work is on. [BE PR here](https://github.com/CenterForOpenScience/osf.io/pull/10499)


[ENG-5019]: https://openscience.atlassian.net/browse/ENG-5019?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
